service/service.go

```diff
@@ -0,0 +1,172 @@
+		APIRouterURL:                cfg.APIRouterURL,
+		CacheUpdateInterval:         cfg.CacheUpdateInterval,
+		EnableNewNavBar:             cfg.EnableNewNavBar,
+		EnableCensusTopicSubsection: cfg.EnableCensusTopicSubsection,
+		CensusTopicID:               cfg.CensusTopicID,
+		IsPublishingMode:            cfg.IsPublishing,
+		Languages:                   cfg.SupportedLanguages,
+		ServiceAuthToken:            cfg.ServiceAuthToken,
+	}
+
+	// Create service initialiser and an error channel for service errors
+	svcErrors := make(chan error)
+
+	cacheService, _ := cacheHelper.Init(ctx, cacheConfig)
+	cacheService.RunUpdates(ctx, svcErrors)
+
+	// Initialise router
+	r := mux.NewRouter()
+	if cfg.OtelEnabled {
+		r.Use(otelmux.Middleware(cfg.OTServiceName))
+	}
+	middleware := []alice.Constructor{
+		renderror.Handler(clients.Renderer),
+	}
+	newAlice := alice.New(middleware...).Then(r)
+	routes.Setup(ctx, r, cfg, clients, cacheService)
+	svc.Server = serviceList.GetHTTPServer(cfg.BindAddr, newAlice)
+
+	return nil
+}
+
+// Run starts an initialised service
+func (svc *Service) Run(ctx context.Context, svcErrors chan error) {
+	log.Info(ctx, "Starting service", log.Data{"config": svc.Config})
+
+	// Start healthcheck
+	svc.HealthCheck.Start(ctx)
+
+	// Start HTTP server
+	log.Info(ctx, "Starting server")
+	go func() {
+		if err := svc.Server.ListenAndServe(); err != nil {
+			log.Fatal(ctx, "failed to start http listen and serve", err)
+			svcErrors <- err
+		}
+	}()
+}
+
+// Close gracefully shuts the service down in the required order, with timeout
+func (svc *Service) Close(ctx context.Context) error {
+	log.Info(ctx, "commencing graceful shutdown")
+	ctx, cancel := context.WithTimeout(ctx, svc.Config.GracefulShutdownTimeout)
+	hasShutdownError := false
+
+	go func() {
+		defer cancel()
+
+		// stop healthcheck, as it depends on everything else
+		log.Info(ctx, "stop health checkers")
+		svc.HealthCheck.Stop()
+
+		// stop any incoming requests
+		if err := svc.Server.Shutdown(ctx); err != nil {
+			log.Error(ctx, "failed to shutdown http server", err)
+			hasShutdownError = true
+		}
+	}()
+
+	// wait for shutdown success (via cancel) or failure (timeout)
+	<-ctx.Done()
+
+	// timeout expired
+	if ctx.Err() == context.DeadlineExceeded {
+		log.Error(ctx, "shutdown timed out", ctx.Err())
+		return ctx.Err()
+	}
+
+	// other error
+	if hasShutdownError {
+		err := errors.New("failed to shutdown gracefully")
+		log.Error(ctx, "failed to shutdown gracefully ", err)
+		return err
+	}
+
+	log.Info(ctx, "graceful shutdown was successful")
+	return nil
+}
+
+func (svc *Service) registerCheckers(ctx context.Context, c routes.Clients) (err error) {
+	hasErrors := false
+
+	if err = svc.HealthCheck.AddCheck("Feedback API", c.FeedbackAPI.Checker); err != nil {
+		hasErrors = true
+		log.Error(ctx, "failed to add feedback API checker", err)
+	}
+
+	if hasErrors {
+		return errors.New("Error(s) registering checkers for healthcheck")
```

Comment:
The error string should not begin with a capital letter.
```suggestion
		return errors.New("error(s) registering checkers for healthcheck")
```

Reply:
Done